src/main/scala/cromwell/binding/values/WdlFile.scala

```diff
@@ -21,11 +22,16 @@ case class WdlFile(value: String) extends WdlPrimitive {
   }
 
   override def equals(rhs: WdlValue): Try[WdlBoolean] = rhs match {
-    case r: WdlFile => Success(WdlBoolean(value.equals(r.value)))
-    case r: WdlString => Success(WdlBoolean(value.toString.equals(r.value)))
+    case r: WdlFile => Success(WdlBoolean(value.equals(r.value) && globbiness.equals(r.globbiness)))
+    case r: WdlString => Success(WdlBoolean(value.toString.equals(r.value.toString) && globbiness == GlobOrNot.NotGlob))
     case _ => invalid(s"$value == $rhs")
   }
 
   override def toWdlString = "\"" + value.toString + "\""
   override def valueString = value.toString
 }
+
+object GlobOrNot extends Enumeration {
```

Comment:
@cjllanwarne you mentioned in the other PR that you weren't able to achieve this via subtyping. Did you try making it into an ADT style structure? e.g. `sealed trait WdlFile` and then children for glob/unglob?